.github/workflows/test.yml

```diff
@@ -1,6 +1,10 @@
 name: "CI"
 
-on: [push, pull_request]
+on:
```

Comment:
Wouldn't `push` have been alright to? 

I think the difference would be:
- `push` -> you need to be able to push to our repo
- `pull_request` -> anyone can create a PR to our repo

Reply:
Good point. So the way it is right now means that the pipeline will run on:
- `push` events to the `main` branch
- `pull_request` events (`opened`, `synchronize` and `reopened` [by default](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request))

This means that other branches (besides `main`) would not run the pipeline until a Pull Request is open. So I think we can either enable to all the branches and your suggestion for `push` would suffice or leave it as it is.

Reply:
> all the branches
this would be all branches **in our repo**. As we normally don't create random branches and push them I don't think this would be an issue.

I do not have a strong opinion and I am fine either way. But I would keep that in mind as this repo will contain security critical secrets.

Reply:
For what I tested on the services no one inside our organization will have access to the secrets, even if they can open a PR. Only way for people external to access the secrets would be allowing them as collaborators on the repo

Reply:
It seems that [`pull_request_target`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target) has access to secrets even when triggered by forks (which was not what was applied here).

But maybe converting it to just `push` simplifies it.

Reply:
Yes, but we were using `pull_request`, right?

Reply:
> Yes, but we were using `pull_request`, right?

Yes 👍